packages/services/api/src/modules/organization/providers/organization-access-tokens-cache.ts

```diff
@@ -0,0 +1,69 @@
+import { BentoCache, bentostore } from 'bentocache';
+import { memoryDriver } from 'bentocache/build/src/drivers/memory';
+import { redisDriver } from 'bentocache/build/src/drivers/redis';
+import { Inject, Injectable, Scope } from 'graphql-modules';
+import Redis from 'ioredis';
+import type { DatabasePool } from 'slonik';
+import { Logger } from '../../shared/providers/logger';
+import { PG_POOL_CONFIG } from '../../shared/providers/pg-pool';
+import { REDIS_INSTANCE } from '../../shared/providers/redis';
+import { findById, type OrganizationAccessToken } from './organization-access-tokens';
+
+/**
+ * Cache for performant OrganizationAccessToken lookups.
+ */
+@Injectable({
+  scope: Scope.Singleton,
+  global: true,
+})
+export class OrganizationAccessTokensCache {
+  private findById: ReturnType<typeof findById>;
+  private cache: BentoCache<{ store: ReturnType<typeof bentostore> }>;
+
+  constructor(
+    @Inject(REDIS_INSTANCE) redis: Redis,
+    @Inject(PG_POOL_CONFIG) pool: DatabasePool,
+    logger: Logger,
+  ) {
+    this.findById = findById({ pool, logger });
+    this.cache = new BentoCache({
+      default: 'organizationAccessTokens',
+      stores: {
+        organizationAccessTokens: bentostore()
+          .useL1Layer(
+            memoryDriver({
+              maxItems: 10_000,
```

Comment:
Should this be configurable? Why 10k?

Reply:
10k is a magic number I choose that seems big enough for the beginning. 😅
Making it configurable rn might be an overkill. However, I think we should monitor the numbers and adjust if needed.

Reply:
Consider adding some more monitoring for this then. E.g. It might be nice to know if this ever exceeds the threshold.
Or are there other ways we could easily determine this is a bottleneck?

Reply:
@jdolle I wanted to add https://bentocache.dev/docs/plugin-prometheus here to have more insights.

Reply:
That looks promising. I like that they provide a grafana configuration. :)